envoy/api/v2/route/route.proto

```diff
@@ -803,7 +814,36 @@ message HeaderMatcher {
   // * The regex *\d{3}* matches the value *123*
   // * The regex *\d{3}* does not match the value *1234*
   // * The regex *\d{3}* does not match the value *123.456*
-  google.protobuf.BoolValue regex = 3;
+  google.protobuf.BoolValue regex = 3 [deprecated = true];
+
+  // [#not-implemented-hide:]
+  // Specifies how the header match will be performed to route the request.
+  // If header_match_specifier is absent, a request that has the name header will match, regardless
+  // of the header's value.
+  oneof header_match_specifier {
+    // [#not-implemented-hide:]
+    // If specified, header match will be performed based on the value of the header.
+    string value_match = 4;
```

Comment:
nit: exact_match, prefix_match, regex_match (or get rid of _match). Prefix match is pretty helpful as well

Reply:
@rshriram  I went the _match suffix to avoid clashes with the existing value, regex fields in this message, which will be deprecated once people switch to this oneof match specifier. and named this range_match for consistency. 
Can change the value_match to exact_match so that we have scope to implement prefix_match later (I am not planning to add that as part of this PR).